### ISSUE TITLE
instagram.com

### ISSUE THREAD

**Issue by in4u** [uBlockOrigin/uAssets#4959]
### URL(s) where the issue occurs

`https://www.instagram.com/p/BkQzGmfjzGq/`

### Describe the issue

The video player controls are hidden / disabled. There is no way to control volume, seek, etc.

### Versions

- Browser/version: Firefox 65.0.1
- uBlock Origin version: 1.18.4

### Settings

Annoyances filter

### Notes

This is not a problem for IGTV or long-form videos. See `https://www.instagram.com/tv/BkQjCfsBIzi/` for example. For normal videos, using uBO's element zapper twice over the video and then right clicking it to enable `Show Controls` setting solves this problem.

**Comment by okiehsch** [comment]
I can't see the video player controls even if I disable `uBO`, seems like a site issue.

**Comment by in4u** [comment]
uBO is not the cause of this problem. It's an annoyance which if fixable via uBO's syntax could be included in the annoyances filter.

**Comment by krystian3w** [comment]
Also no exsist on Chromium.

**Comment by krystian3w** [comment]
You need use 2 times "zapper" :zap:  and unlock by right click.

**Comment by in4u** [comment]
A possible way to resolve this issue could be to remove the two overlay elements and add `controls` attribute to the `video` element. If uBO's extended syntax supports adding an attribute or changing property value this should be doable.

**Comment by yourduskquibbles** [comment]
`##[href^="javascript:"]:has-text(Control)` seems to remove the first overlay, I can't come up with a great way to remove the second overlay that won't rely on the class name not changing from `##.oujXn`

**Comment by uBlock-user** [comment]
The video element is not present in the inline-script and this a site-design issue, they have their own js plugins which load the video, so currently there's no fix.

**Comment by in4u** [comment]
Okay, since the reason for closure is `wontfix` instead of `cantfix`, can you please share the rules which I may use in my own filter list at least? More specifically, how to add `controls` attribute to the `video` element or change `video.controls` property to `true`?

**Comment by krystian3w** [comment]
Instagram only compress class "names", no randomize similar on `Wirtualna Polska` (change after 1 day or after any refresh page).

And uBO do not have scriplet to add attribs (dangerous).

**Comment by uBlock-user** [comment]
> More specifically, how to add `controls` attribute to the `video` element

You need a `set-attribute` scriptlet for that. uBO resources doesn't have that. I did make one for myself, but the problem is that the CSS selector for video element i.e.  ##video for setting the attribute controls doesn't work. Need a exact CSS selector to make it work.

**Comment by krystian3w** [comment]
Strange limitation similar when I try remove `class/style from `body/html` with `remove-attr.js`

**Comment by in4u** [comment]
The rule `instagram.com##:xpath(//video[not(@controls)]/../../../following-sibling::*)` removes both the overlay elements. Just have to figure out a way of toggling the `controls` attribute.

@uBlock-user Is that a bug specific to your scriptlet or something to do with uBO's implementation? How to make it work in this case?

**Comment by uBlock-user** [comment]
> Is that a bug specific to your scriptlet

not a bug as my scriptlet is based on the `remove-attr.js` scriptlet and it works for everything else, it's just that  `video` as the CSS selector doesn't work like it does with `html/body`.

**Comment by krystian3w** [comment]
What does not works?

> `autostrada-a2.pl##+js(remove-attr.js, class, body)`

https://github.com/MajkiIT/polish-ads-filter/blob/79f948049862c5f43eaa04ce8d6b551823c98385/cookies_filters/cookies_uB_AG.txt#L41 - this is works perfect on my side.

**Comment by in4u** [comment]
@uBlock-user I tested `remove-attr.js` for `video` element on [this](https://www.w3schools.com/html/html5_video.asp) page and it does seem to work. If I replace `DOMContentLoaded` in [your scriptlet](https://github.com/uBlock-user/uBO-Scriptlets/blob/e41da50ec0617ce43f25ca2e6fc047cd30a85fec/scriptlets.txt#L173) with `load` then the attribute gets set successfully but it causes my element hiding rule to fail. Any idea how to work around it?

**Comment by krystian3w** [comment]
Video do not have stable class / stable attrib?

**Comment by krystian3w** [comment]
```
var myVideo = document.getElementsByClassName("tWeCl")[0];
myVideo.controls = 'true';
```

Works for my end.

**Comment by in4u** [comment]
@krystian3w That `xpath` rule just hides the two overlay elements. You need to click the video once to play it. Only after that the right-click popup menu shows the control setting.

**Comment by krystian3w** [comment]
Popup show post, in this for me no works xpath, try open from instagram profile:
and use xpath.

![obraz](https://user-images.githubusercontent.com/35370833/53211613-1a63cb80-3642-11e9-95be-cfeff46f069a.png) |
--- |
![obraz](https://user-images.githubusercontent.com/35370833/53211647-40896b80-3642-11e9-9c10-6df66a58369a.png) |

**Comment by krystian3w** [comment]
Popup show post, in this for me no works xpath - excluded i18l (other possible instagram languages), try open from instagram profile:

```
!pop-up
instagram.com##body[class]:not([style]):xpath(//video[not(@controls)]/../../following-sibling::*)
!direct links
instagram.com##body[class=""][style]:xpath(//video[not(@controls)]/../../following-sibling::*)
```

https://www.instagram.com/instagram/

and use xpath.

![obraz](https://user-images.githubusercontent.com/35370833/53211613-1a63cb80-3642-11e9-95be-cfeff46f069a.png) |
--- |
![obraz](https://user-images.githubusercontent.com/35370833/53211647-40896b80-3642-11e9-9c10-6df66a58369a.png) |
![obraz](https://user-images.githubusercontent.com/35370833/53211747-9a8a3100-3642-11e9-8c15-24674be72059.png) |

**Comment by krystian3w** [comment]
Popup show post - radom see this with dogs :dog:  :dog2:  :poodle:, in this for me no works xpath - excluded i18n (other possible instagram languages), try open from instagram profile:

https://www.instagram.com/p/Bt18m4ugdd-/

```
!pop-up
instagram.com##body[class=""]:not([style]):xpath(//video[not(@controls)]/../../following-sibling::*)
!direct links
instagram.com##body[class=""][style]:xpath(//video[not(@controls)]/../../following-sibling::*)
```

https://www.instagram.com/instagram/

and use xpath.

![obraz](https://user-images.githubusercontent.com/35370833/53211613-1a63cb80-3642-11e9-95be-cfeff46f069a.png) | demo
--- | ---
![obraz](https://user-images.githubusercontent.com/35370833/53211647-40896b80-3642-11e9-9c10-6df66a58369a.png) | Why no works?
![obraz](https://user-images.githubusercontent.com/35370833/53213335-cad4ce00-3648-11e9-9dbf-ae45c23c6861.png) | Why works?
![obraz](https://user-images.githubusercontent.com/35370833/53211747-9a8a3100-3642-11e9-8c15-24674be72059.png) | still good?

**Comment by uBlock-user** [comment]
> If I replace `DOMContentLoaded` in [your scriptlet](https://github.com/uBlock-user/uBO-Scriptlets/blob/e41da50ec0617ce43f25ca2e6fc047cd30a85fec/scriptlets.txt#L173) with `load` then the attribute gets set successfully

@in4u good find! I think the video element is dynamically added by instagram after the page load, this is why `DOMContentLoaded` event won't work, but `load` works perfectly as expected. This is the filter to use if you're using my scriptlet.

`instagram.com##+js(set-attr.js, controls, controls, video)`

> but it causes my element hiding rule to fail. Any idea how to work around it?

which rule ?

**Comment by in4u** [comment]
@krystian3w Thanks for providing the example link. Indeed there was one less `div` on that page. Since the no. of divs possibly vary, maybe `instagram.com##a[href="javascript:;"][role="button"]:xpath(.|preceding-sibling::*[1])` would be a better rule. Do you see any problems with it?

@uBlock-user My new rule above has no problem with your scriptlet. For the previous rule, either there was something wrong with my syntax or uBO's implementation or the scriptlet. I am still scratching my head over this and will open an issue at your repo if I am unable to sort it out. Also, for dynamically loaded elements even `remove-attr.js` fails to work. Haven't tested if `load` resolves this problem. Regarding that, where should issues related to uBO's default scriptlets be reported - here or over at https://github.com/uBlockOrigin/uBlock-issues?

**Comment by uBlock-user** [comment]
> Also, for dynamically loaded elements even `remove-attr.js` fails to work. Haven't tested if `load` resolves this problem.

Yes that's bound to happen, dynamically generated elements in the DOM need to be accounted by `MutationObserver` and apply the patch as detected, as a side effect of doing that, performance will be affected.

**Comment by in4u** [comment]
Ah, alright. One more thing... how to force uBO to fetch an updated scriptlet file? Currently I am testing by changing user resource URL in the settings every time in order to see instant results. There is an option to invalidate cache and force update of individual filter lists. Is there any way to do this for the scriptlet file?

**Comment by uBlock-user** [comment]
> how to force uBO to fetch an updated scriptlet file?

Force update uBlock Filters

**Comment by krystian3w** [comment]
No works in my Firefox 65:

> `instagram.com##+js(set-attr.js, controls, controls, video)`

![obraz](https://user-images.githubusercontent.com/35370833/53268267-e0e49c00-36e5-11e9-988b-73f4a3548fca.png) |
--- |

**Comment by in4u** [comment]
Strange! Which rules are you using @krystian3w? I have the following two rules and it's working for me:

`instagram.com##a[href="javascript:;"][role="button"]:xpath(.|preceding-sibling::*[1])`
`instagram.com##+js(set-attr.js, controls, controls, video)`

Are the two overlay elements removed in your case? If no then please share an example page. If yes then maybe you should force an update of the scriptlet as suggested by @uBlock-user.

**Comment by krystian3w** [comment]
yes I used your and @uBlock-user  version.

Only cut "overlays".

**Comment by in4u** [comment]
What happens when you click on the video?

**Comment by krystian3w** [comment]
Vide play, no see video bar, script no unlock on videos:

https://www.instagram.com/p/BkQzGmfjzGq/
https://www.instagram.com/p/Bt18m4ugdd-/

Maybe blocked by CSP (`security.csp.enable` = false).

**Comment by in4u** [comment]
It works in my case even after setting `security.csp.enable` as `false`.

I guess only @uBlock-user can help out here. Maybe you should cut the ribbon over at https://github.com/uBlock-user/uBO-Scriptlets/issues and open an issue there!

**Comment by krystian3w** [comment]
IMO no possible easy override this with uBO only.

**Comment by ghajini** [comment]
@mapx-
any filter to remove this login annoyance

`https://www.instagram.com/accounts/login/?next=%2Fp%2FB3kHvgehLZV%2Fcomments%2F&source=post_comment_view_all`

Or go here `https://www.instagram.com/p/B3kHvgehLZV/`
And click ***view all comments***

Screenshot=

`https://prnt.sc/pjagpu`

**Comment by mapx-** [comment]
I don't think it's possible (pages order established on server side - it's not only an overlay as I can see ..)

**Comment by krystian3w** [comment]
Tiktok / music.ly have similar element:

```
tiktok.com##.video-card-container > .image-card > .video-card > video ~ *:not(.mute-icon)
```

**Comment by llacb47** [comment]
Why are all of your earlier comments gone? 👀

**Comment by krystian3w** [comment]
- O que essazinha tá fazendo aqui?
- Nos encontramos de novo, né?
